tests/test_encoder.js

```diff
@@ -0,0 +1,161 @@
+    'NbsEncoder() constructor throws with incorrect type for path argument'
+  );
+});
+
+test('NbsEncoder write throws for invalid arguments', () => {
+  const writePath = path.join(samplesDir, 'writeTest.nbs');
+  const encoder = new NbsEncoder(writePath);
+
+  assert.throws(
+    () => {
+      encoder.write();
+    },
+    /missing argument `packet`: provide a packet to write to the file/,
+    'NbsEncoder write() throws without packet argument'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({});
+    },
+    /invalid type for argument `packet`: expected object with `timestamp` key/,
+    'NbsEncoder write() throws without timestamp in packet'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: { seconds: 1897, nanos: 0 },
+      });
+    },
+    /invalid type for argument `packet`: expected object with `type` key/,
+    'NbsEncoder write() throws without type in packet'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: { seconds: 1897, nanos: 0 },
+        type: pingType,
+      });
+    },
+    /invalid type for argument `packet`: expected object with `subtype` key/,
+    'NbsEncoder write() throws without subtype in packet'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: { seconds: 1897, nanos: 0 },
+        type: pingType,
+        subtype: 0,
+      });
+    },
+    /invalid type for argument `packet`: expected object with `payload` key/,
+    'NbsEncoder write() throws without payload in packet'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: 'string',
+        type: pingType,
+        subtype: 0,
+        payload: Buffer.from('ping.699', 'utf8'),
+      });
+    },
+    /invalid type for argument `packet`: error in `timestamp`: expected positive number or BigInt or timestamp object/,
+    'NbsEncoder write() throws with incorrect type for timestamp'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: { wrongKey: 1 },
+        type: pingType,
+        subtype: 0,
+        payload: Buffer.from('ping.699', 'utf8'),
+      });
+    },
+    /invalid type for argument `packet`: error in `timestamp`: expected object with `seconds` and `nanos` keys/,
+    'NbsEncoder write() throws for incorrect keys in timestamp'
+  );
+
+  assert.throws(
+    () => {
+      encoder.write({
+        timestamp: { seconds: 'string', nanos: 'string' },
+        type: pingType,
+        subtype: 0,
+        payload: Buffer.from('ping.699', 'utf8'),
+      });
+    },
+    /invalid type for argument `packet`: error in `timestamp`: `seconds` and `nanos` must be numbers/,
+    'NbsEncoder write() throws with incorrect types for timestamp properties'
+  );
+
+  // Delete file written to
+  fs.unlinkSync(writePath);
+  fs.unlinkSync(writePath + '.idx');
+});
+
+test('Packets written by NbsEncoder can be read by NbsDecoder', () => {
+  const writePath = path.join(samplesDir, 'writeTest.nbs');
```

Comment:
Maybe for these tests take advantage of the https://www.npmjs.com/package/tmp package to make a temporary director so it'll be cleaned up automatically on close

Reply:
I don't think the extra package (13 if counting transitive tendencies!) is worth it here.

I've left a review comment on the change with an alternative below.